### PR TITLE
Add cross link on beacon chain

### PR DESCRIPTION
**Pull request by rlan35**
tested locally with 2 shards.

### CODE REVIEW

File: api/proto/node/node.go [hunk 1]
```diff
@@ -165,6 +165,17 @@ func ConstructBlocksSyncMessage(blocks []*types.Block) []byte {
 	return byteBuffer.Bytes()
 }
 
+// ConstructCrossLinkHeadersMessage constructs cross link header message to send blocks to beacon chain
```
Comment (chaosma) [inline api/proto/node/node.go]: send blocks => send block headers

File: core/blockchain.go [hunk 2]
```diff
@@ -1121,8 +1121,20 @@ func (bc *BlockChain) InsertChain(chain types.Blocks) (int, error) {
 					header.Logger(utils.Logger()).Warn().Err(err).Msg("cannot store shard state")
 				}
 			}
+			if len(header.CrossLinks) > 0 {
+				crossLinks := &types.CrossLinks{}
+				err := rlp.DecodeBytes(header.CrossLinks, crossLinks)
+				if err != nil {
+					header.Logger(utils.Logger()).Warn().Err(err).Msg("[insertChain] cannot parse cross links")
```
Comment (chaosma) [inline core/blockchain.go]: should we return here?

File: api/proto/node/node.go (hunk 1 above)
Comment (rlan35) [reply api/proto/node/node.go]: Done

File: core/blockchain.go (hunk 2 above)
Comment (rlan35) [reply core/blockchain.go]: Done.

**Pull request merged**